appcontainer/ImpersonationBarModel.js

```diff
@@ -33,6 +33,7 @@ export class ImpersonationBarModel {
     @action
     show() {
         throwIf(!XH.identityService.canImpersonate, 'User does not have right to impersonate.');
+        throwIf(!XH.getConf('xhEnableImpersonation', false), 'Impersonation is disabled for this app.');
```

Comment:
Thanks for getting this change in! 

What do you think about baking this into `XH.identityService.canImpersonate`?  It seems to me that getter should take the config into account on its own. 

That would also prevent `ImpersonationBar` from rendering entirely and installing its hotkey handler. It's a little mixed how it works now (i.e. if you are not a Hoist admin, bar never renders, but if you and impersonation disabled, bar renders but throws if you attempt to open.

Even more importantly, it would stop mobile from displaying a non-functional button in the app menu - we need to fix that regardless:

https://github.com/xh/hoist-react/blob/19234a5a213b460af2d3ce34e44d113658904fa0/mobile/cmp/header/AppMenuModel.js#L65

(Mobile adds a button as you don't have hotkeys)

Reply:
And the logic for whether or not impersonation is allowed should be baked into identityService.